datafusion/core/src/datasource/statistics.rs

```diff
@@ -33,101 +36,95 @@ pub async fn get_statistics_with_limit(
     limit: Option<usize>,
 ) -> Result<(Vec<PartitionedFile>, Statistics)> {
     let mut result_files = vec![];
+    // These statistics can be calculated as long as at least one file has them.
```

Comment:
Another pattern, if we wanted to encapsulate this code / make it easier to unit test could be to put these fields in a "builder" style pattern. This way is fine too I just figured I would point it out

```rust
let builder = FileStatisticsBuilder::new(file_schema);

...
    if let Some(first_file) = all_files.next().await {
        let (file, file_stats) = first_file?;
        result_files.push(file);

        // First file, we set them directly from the file statistics.
        builder.add_file(
            file_stats,
        );
...

    let mut statistics: Statistics = builder.build()

...

```